cesium/internal/meta/meta.go

```diff
@@ -29,16 +30,19 @@ func ReadOrCreate(fs xfs.FS, ch core.Channel, ecd binary.EncoderDecoder) (core.C
 	if err != nil {
 		return ch, err
 	}
-	if !exists {
-		if ch.Key == 0 {
-			return ch, errors.Wrap(
-				validate.Error,
-				"[meta] - a channel is required when creating a new database",
-			)
-		}
-		return ch, Create(fs, ecd, ch)
+	if exists {
+		return Read(fs, ecd)
 	}
-	return Read(fs, ecd)
+
+	if ch.DataType == telem.UnknownT {
```

Comment:
We should validate other properties of the channel here no? Like checking that the key is not zero. Make sure we also check other properties that need to be well defined that I did not mention. Also, I feel like we should move this check into the `Create` function no?